### ISSUE TITLE
Error when calling skaffold delete on projects with dependent modules

### ISSUE THREAD

**Issue by darkolakovic** [GoogleContainerTools/skaffold#7203]
### Expected behavior
Cleaning up modules should start from the dependent modules followed by the modules previous modules depend on.

### Actual behavior
Cleaning up modules starts from the modules other modules depend on. Therefore cleaning up dependent modules throws errors since the modules other depend on are already deleted.

### Information

- Skaffold version: v1.36.1
- Operating system: MacOS Montrey 12.1
- Installed via: Homebrew
- Contents of skaffold.yaml:

```yaml
apiVersion: skaffold/v2beta22
kind: Config
requires:
  - configs:
      - "cert-manager"
deploy:
  helm:
    releases:
      - name: custom-cert
        chartPath: ./CustomCert
        namespace: custom-cert
        createNamespace: true
        recreatePods: false
        skipBuildDependencies: false
        useHelmSecrets: false
        wait: false
    
---

apiVersion: skaffold/v2beta22
kind: Config
metadata:
  name: cert-manager
deploy:
  helm:
    releases:
      - name: cert-manager
        remoteChart: cert-manager
        version: v1.7.1
        repo: https://charts.jetstack.io
        overrides:
          installCRDs: true
        namespace: cert-manager
        createNamespace: true
        recreatePods: false
        skipBuildDependencies: false
        useHelmSecrets: false
        wait: false
```

### Steps to reproduce the behavior

1. Download project from: [ModuleDelete.zip](https://github.com/GoogleContainerTools/skaffold/files/8303663/ModuleDelete.zip)
2. `skaffold dev`
3. After deployment is stabilised press Control+C to exit so cleanup process can start
4. You will see that first depending module is uninstalled, after which there is an error thrown during the uninstallation of module using it: 
`Cleaning up...
release "cert-manager" uninstalled
Error: uninstallation completed with 1 error(s): unable to build kubernetes objects for delete: unable to recognize "": no matches for kind "ClusterIssuer" in version "cert-manager.io/v1"
WARN[0066] deployer cleanup:exit status 1                subtask=-1 task=DevLoop`

Log:
[skaffold_module_vdebug.log](https://github.com/GoogleContainerTools/skaffold/files/8303746/skaffold_module_vdebug.log)

**Comment by tejal29** [comment]
Thanks for this issue. Seems like a reasonable approach.

**Comment by ericzzzzzzz** [comment]
seems like this has already been fixed.. I'll link the fix pr later.

**Comment by ericzzzzzzz** [comment]
the same as #7284 which should be fixed by #7925 in skaffold v2
